### PR TITLE
Add: nimclient package module for AIX

### PR DESCRIPTION
**Pull request by nickanderson**
This module provides basic functionality for using nimclient as a means
to ensure packages are either present or absent. It does not support
listing package updates available or provide any special caching.

### CODE REVIEW

**Comment by nickanderson** [comment]
ping @kacf PR as requested.

File: modules/packages/nimclient [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  if [ -n "$options" ]; then
+    eval $options
+  fi
+done
+
+# Set up mock environment if necessary
+# This is not well developed as I don't have continuous access to aix and nim
+# nor am I an expert
+CFENGINE_TEST_NIMCLIENT_MOCK=false
+if [ -n "$CFENGINE_TEST_PKGSRC_MOCK" ]; then
```
Comment (kacf) [inline modules/packages/nimclient]: These two variables are different, but I assume they should be the same? In that case the `if [ -n` check works only by accident, you should probably correct this to:

```
CFENGINE_TEST_NIMCLIENT_MOCK=false
if [ "$CFENGINE_TEST_NIMCLIENT_MOCK" = "true" ]; then
```

File: modules/packages/nimclient [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    list_installed_packages() {
+        cat ./mock_lslpp_Lc
+     }
+    nimclient_showres() {
+        # This lists the AVAILABLE packages in the nim repo
+        cat ./mock_nimclient_showres
```
Comment (kacf) [inline modules/packages/nimclient]: The mock files are not in the correct folder for this to work. It should be: `cat ../../tests/unit/mock_lslpp_Lc` and `cat ../../tests/unit/mock_nimclient_showres`.

**Comment by kacf** [comment]
Looks good apart from the comments.

You might also want to think about turning those mock files into a unit test that runs automatically. It can simply be a shell script that calls the package module with the mock environment variable set, and checks for output. But this is very platform specific, so I won't hold it back if you don't have the bandwidth for that right now. Just let me know.

File: modules/packages/nimclient (hunk 1 above)
Comment (nickanderson) [inline modules/packages/nimclient]: ack

File: modules/packages/nimclient (hunk 2 above)
Comment (nickanderson) [inline modules/packages/nimclient]: ack

**Comment by nickanderson** [comment]
@kacf updated per your comments, but i did not make any unit test shell script for this. I will skip it if I can.

**Comment by kacf** [comment]
Sure. I think we should not hold it back because of that. Looks good now!

**Comment by kacf** [comment]
Merged manually after adding changelog entry.

**Pull request closed without merging**